scarb/src/compiler/compilation_unit.rs

```diff
@@ -145,6 +184,45 @@ impl CompilationUnit {
     }
 }
 
+impl CompilationUnitAttributes for ProcMacroCompilationUnit {
+    fn main_package_id(&self) -> PackageId {
+        self.main_package_id
+    }
+    fn components(&self) -> &[CompilationUnitComponent] {
+        &self.components
+    }
+
+    fn digest(&self) -> String {
+        let mut hasher = StableHasher::new();
+        self.main_package_id.hash(&mut hasher);
+        for component in &self.components {
+            component.hash(&mut hasher);
+        }
+        hasher.finish_as_short_hash()
+    }
+}
+
+impl CairoCompilationUnit {
+    pub fn core_package_component(&self) -> Option<&CompilationUnitComponent> {
+        // NOTE: This uses the order invariant of `component` field.
+        if self.components.len() < 2 {
+            None
+        } else {
+            let component = &self.components[1];
+            assert!(component.package.id.is_core());
+            Some(component)
+        }
+    }
+
+    pub fn target_dir(&self, ws: &Workspace<'_>) -> Filesystem {
+        ws.target_dir().child(self.profile.as_str())
+    }
+
+    pub fn is_cairo_plugin(&self) -> bool {
+        self.target().is_cairo_plugin()
+    }
```

Comment:
Can you please add a doc comment stating why does this method exist for cairo compilation unit, given that we have a different enum branch to represent plugins? As this looks contradictory from the first glimpse.

Reply:
Good point, removed it.